### ISSUE TITLE
Linker Error on Windows - unresolved _snprintf

### ISSUE THREAD

**Issue by hansjoachimknobloch** [cisco/libest#91]
What happened to the line

#define snprintf _snprintf

in src/est/est_client.c (was line #27)?

When compiling est.dll under Windows using VC 2013, I get a linker error about the unresolved reference _snprintf.
After re-adding the above line, the libary is sucessfully built and the example estclient using it works fine against the public test server at testrfc7030.com.

**Comment by rpb5bnc** [comment]
Sorry for the delay.

That #define was needed when building with older versions of Microsoft visual studio.  I believe that MS never had a fully standard version of snprintf() in their C runtime library.  Aliasing snprintf() to _snprintf() allowed us to use the MS specific one without any issues.  Starting with MSVC 14.00 Microsoft finally included a standard version of snprintf().  The internal version for 14.00 is 1900.

In an effort to stay current, when preparing for a release we will attempt to perform a full regression test using the latest dependent packages and latest tool chains.  The latest version of MS C includes the snprintf() in stdio.h and causes conflicts with our aliasing of the function.

You can leave the #define in if it works for you.  A change was pushed to master to allow the code to build in both older and newer MS C tool chains.

Sorry for the hiccup.

**Comment by hansjoachimknobloch** [comment]
Thanks, Pete!

Your explanation is sound, however:
- According to libest/README the newst Visual Studio toolchain to build libest and the Examples is Visual Studio 2013, which is what I used.
- In numerous other source files, e. g. libest/src/est/est_client_proxy.c, the
```
#ifdef WIN32
#   define snprintf _snprintf
#endif
```
statements are still present.

Therefore I suspect the change to allow for allow for the use of newer Visual C toolchains has not been completely merged into master.

**Comment by rpb5bnc** [comment]
Hello,

I appreciate the feedback on this. 

It appears the README is now out of date given that we're now using VS 2017.  When we get a chance we'll update it.

When we prepared for the last release of libEST a Windows based regression run was made and we used MS Visual Studio 2017.  At that time there was only one compiler error and it pointed to this one #define in est_client.c  The person handling this regression test effort simply removed the line and the code then compiled clean and the full regression test completed successfully.  The change probably should have been a bit more robust to allow for the continued use of older versions of MS Visual Studio.  That's the change that was just pushed to master.

As for the other continued occurrences of the same #define of snprintf in other .c files and why they do not cause the same duplicate symbol error as the one in est_client.c did, I came to the conclusion that it has something to do with the way MS has defined their new prototype for snprintf() combined with the order and combination of other header files being included in these other .c files containing the #define to alias snprintf().  I gave a bit of time to investigate, but didn't come up with anything conclusive.  The bottom line for us was, the code is now compiling and running correctly with both VS 2013 and VS 2017.

Thanks again for the input,
Pete
